### PR TITLE
optional tie-in argument to lax.create_token when omnistaging is on

### LINKED ISSUE #5218: remove tie-in operands under omnistaging for primitives that don't need them
With omnistaging enabled, any primitive such as `create_token`, and its corresponding `lax` function, no longer needs to take a tie-in argument for data dependence.

Removing `create_token`'s operands also enforces that the primitive can't be linear, so it can't be transposed, and the system shouldn't try to transpose it. As @mattjj points out, a nullary primitive wouldn't appear in a jaxpr to be transposed because partial evaluation will never pick it up. This ought to address [this comment in #4292](https://github.com/google/jax/issues/4292#issuecomment-746289386).

### PR DESCRIPTION
**Pull request by froystig**
fixes #5218

### CODE REVIEW

File: jax/_src/lax/lax.py [hunk 1]
```diff
@@ -5898,21 +5898,27 @@ def _stop_gradient_batch_rule(batched_args, batch_dims):
...
-def create_token(x):
+def create_token(x=None):
   """Creates an XLA token value with no preconditions for sequencing effects.
 
   Experimental.
 
   Args:
-    x: a dummy argument used to tie the CreateToken operator into a trace. The
-       value of `x` is ignored.
+    x: Optional dummy argument used to tie the CreateToken operator into a
```
Comment (mattjj) [inline jax/_src/lax/lax.py:5907]: Let's be even more bold here: just say the argument is ignored and nothing else. Consider the docstring of `lax.tie_in` itself!

WDYT about a docstring of:

```python
def create_token(_=None):
  """Creates an XLA token value with no preconditions for sequencing effects.

  Experimental.

  The argument is ignored. It exists for backward compatibility.
  """
```

**Review by mattjj — APPROVED** [review]
LGTM! One suggestion on the docstring.

File: jax/_src/lax/lax.py (hunk 1 above)
Comment (froystig) [reply jax/_src/lax/lax.py]: Good call.

**Comment by froystig** [comment]
Also fixes #4292

**Pull request merged**